app/lib/system_admin_constraint.rb

```diff
@@ -8,7 +8,17 @@ def matches?(request)
 private
 
   def system_admin?(request)
-    signin_user = DfESignInUser.load_from_session(request.session)
+    signin_user =
+      if otp_feature_enabled?
+        OtpSignInUser.load_from_session(request.session)
+      else
+        DfESignInUser.load_from_session(request.session)
+      end
```

Comment:
do we want the system admin constraint to have a detailed understanding of OTP and DSI? perhaps there's a case for having this logic in a separate module that provides `sign_in_user`.

obviously a complication here is that this is a constraint and thus the handy things in `application_controller` aren't available.

Reply:
Fair point, I've extracted a service. All it needs is a `session` but this check does entail an extra DB lookup that's necessary in the controller (because `current_user`).

I did also wonder whether it's really necessary to interrogate the feature flag. Maybe it's sufficient to just do: 

```
signin_user = DfESignInUser.load_from_session(request.session) || OtpSignInUser.load_from_session(request.session)